UploadDaemon_Test/Config/ConfigTest.cs

```diff
@@ -408,7 +408,39 @@ public void TestConfigureUploadInterval()
                         targetdir: C:\test1
             ");
 
-            Assert.That(config.UploadIntervalInMinutes, Is.EqualTo(42));
-        }
-    }
+            Assert.That(config.UploadInterval, Is.EqualTo(TimeSpan.FromMinutes(42)));
+        }
+
+        [Test]
+        public void TestConfigurePurgeThresholds()
+        {
+            Config config = Config.Read(@"
+                archivePurgingThresholdsInDays:
+                  uploadedTraces: 1
+                  emptyTraces: 3
+                  incompleteTraces: 7
+                match:
+                    - profiler:
+                        targetdir: C:\test1
+            ");
+
+            Assert.That(config.ArchivePurgingThresholds.UploadedTraces, Is.EqualTo(TimeSpan.FromDays(1)));
+            Assert.That(config.ArchivePurgingThresholds.EmptyTraces, Is.EqualTo(TimeSpan.FromDays(3)));
+            Assert.That(config.ArchivePurgingThresholds.IncompleteTraces, Is.EqualTo(TimeSpan.FromDays(7)));
+        }
+
+        [Test]
+        public void TestConfigurePurgeThresholdDefaults()
+        {
+            Config config = Config.Read(@"
+                match:
+                    - profiler:
+                        targetdir: C:\test1
+            ");
+
+            Assert.That(config.ArchivePurgingThresholds.UploadedTraces, Is.LessThan(TimeSpan.Zero));

```

Comment:
please add a test where only some of the options are specified and verify that the missing options get the correct  default value